dipy/denoise/patch2self.py

```diff
@@ -8,8 +8,7 @@
 linear_model, _, _ = optional_package('sklearn.linear_model')
 
 if not has_sklearn:
-    w = "Scikit-Learn is required to denoise the data via Patch2Self."
-    warn(w)
+    warn(sklearn._msg)
```

Comment:
I don't see where this variable is defined

Reply:
`sklearn` is returned here:
https://github.com/dipy/dipy/blob/8ac88459bdd27aed2fe3f518b60e66b94ab7cd15/dipy/denoise/patch2self.py#L7

See here the method that checked the optional package availability:
https://github.com/dipy/dipy/blob/master/dipy/utils/optpkg.py#L91

I had to use the private `_msg` member since trying to access the attribute as `msg` raises the error,

Looks like the change is not pleasant for some builds, but the changed files' tests passing locally for me (Python 3.8, Ubuntu 20.04), e.g.
```
============================= test session starts ==============================
collecting ... collected 7 items

test_sfm.py::test_design_matrix 
test_sfm.py::test_sfm 
test_sfm.py::test_predict 
test_sfm.py::test_sfm_background 
test_sfm.py::test_sfm_stick 
test_sfm.py::test_sfm_sklearnlinearsolver 
test_sfm.py::test_exponential_iso 

=================== 4 passed, 3 skipped, 1 warning in 1.02s ====================

Process finished with exit code 0
PASSED                                   [ 14%]SKIPPED (We need package sklearn for these fun...) [ 28%]
Skipped: We need package sklearn for these functions, but ``import sklearn`` raised an ImportError
SKIPPED (We need package sklearn for these...) [ 42%]
Skipped: We need package sklearn for these functions, but ``import sklearn`` raised an ImportError
PASSED                                  [ 57%]PASSED                                       [ 71%]PASSED                         [ 85%]SKIPPED (We need package sklearn f...) [100%]
Skipped: We need package sklearn for these functions, but ``import sklearn`` raised an ImportError
```

Reply:
Resolving: https://github.com/dipy/dipy/pull/2480#discussion_r743862977.